### PR TITLE
Clarifies the meaning of the external mutability.

### PR DESCRIPTION
**Pull request by GuillaumeRochat**

### CODE REVIEW

**Comment by KaivoAnastetiks** [comment]
@steveklabnik It fixes #33924 using the last suggestion from @jwatt. Is it clear enough this way? Hopefully I'm doing it the right way. Thanks!

File: src/doc/book/mutability.md [hunk 1]
```diff
@@ -62,8 +62,8 @@ Note that here, the `x` is mutable, but not the `y`.
 # Interior vs. Exterior Mutability
 
 However, when we say something is ‘immutable’ in Rust, that doesn’t mean that
-it’s not able to be changed: we mean something has ‘exterior mutability’. Consider,
-for example, [`Arc<T>`][arc]:
+it’s not able to be changed: we are referring to it's ‘exterior mutability’ that
```
Comment (thiagooak) [inline src/doc/book/mutability.md]: remove the apostrophe `it's` should be `its`

Comment (KaivoAnastetiks) [inline src/doc/book/mutability.md]: Whoops! Thanks for catching it!

**Comment by steveklabnik** [comment]
@KaivoAnastetiks thank you! Looks good.

@bors: r+ rollup

**Pull request merged**